### PR TITLE
Add Pentax K-70 support

### PR DESCRIPTION
**Pull request by nijel**
First contribution, hopefully I've done it correctly.

### CODE REVIEW

**Comment by nijel** [comment]
There is also https://github.com/klauspost/rawspeed/pull/162

File: src/external/adobe_coeff.c [hunk 1]
```diff
@@ -404,6 +404,7 @@ static void dt_dcraw_adobe_coeff(const char *name, float cam_xyz[1][12])
     { "Pentax K-1", { 8827,-2829,-1238,-3611,12203,1549,-897,1687,6291 } },
     { "Pentax K-30", { 5789,-1749,-775,-2618,8062,1233,-642,1125,4278 } },
     { "Pentax K-50", { 6184,-1869,-828,-2618,8062,1233,-639,1120,4259 } },
+    { "Pentax K-70", { 53173/65536,-13613/65536,-8351/65536,-28565/65536,84891/65536,9926/65536,-7148/65536,12814/65536,39610/65536 } },
```
Comment (LebedevRI) [inline src/external/adobe_coeff.c]: Please just use `https://github.com/darktable-org/darktable/blob/master/tools/dngmeta.sh` or `https://github.com/darktable-org/darktable/blob/master/tools/dngmeta.rb`

File: tools/noise/gen-profile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (LebedevRI) [inline tools/noise/gen-profile]: Are you sure?

File: src/external/rawspeed/data/cameras.xml [hunk 3]
```diff
@@ -6727,6 +6727,17 @@
...
+	<Camera make="RICOH IMAGING COMPANY, LTD." model="PENTAX K-70">
+		<ID make="Pentax" model="K-70">Pentax K-70</ID>
+		<CFA width="2" height="2">
+			<Color x="0" y="0">RED</Color>
+			<Color x="1" y="0">GREEN</Color>
+			<Color x="0" y="1">GREEN</Color>
+			<Color x="1" y="1">BLUE</Color>
+		</CFA>
+		<Crop x="0" y="0" width="0" height="0"/>
+		<Sensor black="64" white="16319"/>
```
Comment (LebedevRI) [inline src/external/rawspeed/data/cameras.xml]: These come from `Adobe DNG Converter`?

File: src/external/adobe_coeff.c (hunk 1 above)
Comment (LebedevRI) [inline src/external/adobe_coeff.c]: i.e. it should be `8113, ...`

File: src/external/rawspeed/data/cameras.xml (hunk 3 above)
Comment (LebedevRI) [inline src/external/rawspeed/data/cameras.xml]: Hm, it's `Pentax`, don't they natively write DNG? Is it from that dng?

**Comment by LebedevRI** [comment]
In light of all the comments, could you please just open a new issue (Tracker = camera support) https://redmine.darktable.org/projects/darktable/issues/new and attach the raw image (pef, + dng if camera can write it natively).

File: src/external/rawspeed/data/cameras.xml (hunk 3 above)
Comment (nijel) [inline src/external/rawspeed/data/cameras.xml]: It's from camera DNG.

File: tools/noise/gen-profile (hunk 2 above)
Comment (nijel) [inline tools/noise/gen-profile]: This really shouldn't be in this commit, but it fixes color output from the script.

File: src/external/adobe_coeff.c (hunk 1 above)
Comment (nijel) [inline src/external/adobe_coeff.c]: This is actually what I've used:

```
$ ./tools/dngmeta.sh /tmp/wb/IMGP0768.DNG 
DNG created by : PENTAX K-70 Ver. 1.10 
DNG Illuminant : 21 (should be 21)


$ nano -w src/external/adobe_coeff.c

{ "PENTAX K-70", { 53173/65536,-13613/65536,-8351/65536,-28565/65536,84891/65536,9926/65536,-7148/65536,12814/65536,39610/65536 } },

$ git commit -a -m "adobe_coeff: RICOH IMAGING COMPANY, LTD. PENTAX K-70 support" --author "Michal Cihar <nijel@nutt>"


$ nano -w src/external/rawspeed/data/cameras.xml (mind the tabs)

	<Camera make="RICOH IMAGING COMPANY, LTD.  " model="PENTAX K-70        ">
		<ID make="Ricoh" model="PENTAX K-70        ">PENTAX K-70</ID>
		<CFA width="2" height="2">
			<Color x="0" y="0">RED</Color>
			<Color x="1" y="0">GREEN</Color>
			<Color x="0" y="1">GREEN</Color>
			<Color x="1" y="1">BLUE</Color>
		</CFA>
		<Crop x="0" y="0" width="0" height="0"/>
		<Sensor black="64" white="16319"/>
	</Camera>

NOTE: The default crop exposes the full sensor including garbage pixels, which need to be visually inspected. (negative width/height values are right/bottom crops, which are preferred)

NOTE: Sensor black and white levels sometimes vary based on ISO, please run this tool on raws for each of the camera's supported ISOs

```

**Comment by nijel** [comment]
PEF and DNG are attached to https://redmine.darktable.org/issues/11243

File: src/external/rawspeed/data/cameras.xml (hunk 3 above)
Comment (LebedevRI) [inline src/external/rawspeed/data/cameras.xml]: Ah, even better then

**Comment by nijel** [comment]
I've updated the commit and added WB presets as well.

**Comment by LebedevRI** [comment]
@nijel thank you!

**Pull request merged**